schematic/configuration/dataclasses.py

```diff
@@ -0,0 +1,170 @@
+"""Pydantic dataclasses"""
+
+import re
+from dataclasses import field
+from pydantic.dataclasses import dataclass
+from pydantic import validator, ConfigDict, Extra
+
+# This turns on validation for value assignments after creation
+pydantic_config = ConfigDict(validate_assignment=True, extra=Extra.forbid)
+
+
+@dataclass(config=pydantic_config)
+class SynapseConfig:
+    """
+    config_basename: the basename of the synapse config file
+    manifest_basename: the name of downloaded manifest files
+    master_fileview_id: Synapse ID of the file view listing all project data assets.
+    manifest_folder: name of the folder manifests will be saved to locally
+    """
+
+    config_basename: str = ".synapseConfig"
```

Comment:
This also does not match what you have in the config.yml. In your `config.yml`, you have "asset_store > synapse > config" instead of "config_basename"

Reply:
Same as the above issue.